tests/hub-stack-deploy.bats

```diff
@@ -0,0 +1,156 @@
+#!./bats/bin/bats
+# Copyright (c) 2023 EPAM Systems, Inc.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+gen_stack() {
+    mkdir -p "$1/component"
+    cat > "$1/.env" << EOF
+HUB_DOMAIN_NAME="iamtest"
+HUB_STACK_NAME="iamtest"
+HUB_INTERACTIVE="0"
+HUB_FILES="$1/hub.yaml"
+HUB_STATE="$1/hub.state"
+HUB_ELABORATE="$1/hub.elaborate"
+EOF
+    cat > "$1/hub.yaml" << EOF
+version: 1
+kind: stack
+components:
+- name: dummy1
+  source:
+    dir: component
+- name: dummy2
+  source:
+    dir: component
+parameters:
+- name: hub.stackName
+  fromEnv: HUB_STACK_NAME
+EOF
+
+    cat <<EOF > "$1/component/hub-component.yaml"
+version: 1
+kind: component
+parameters:
+- name: hub.stackName
+  env: WORLD
+EOF
+    cat <<EOF > "$1/component/deploy"
+#!/bin/sh -e
+echo "hello, \$WORLD \$VERB!"
+EOF
+    cat <<EOF > "$1/component/undeploy"
+#!/bin/sh -e
+echo "hello, \$WORLD \$VERB!"
+EOF
+    chmod +x "$1/component/deploy" "$1/component/undeploy"
+}
+
+gen_hubctl_double() {
+    mkdir -p "$1"
+    cat <<EOF > "$1/hubctl"
+#!/bin/sh
+echo ">> hubctl \$@"
+EOF
+    chmod +x "$1/hubctl"
+}
+
+setup() {
+    load 'helpers/bats-support/load'
+    load 'helpers/bats-assert/load'
+    # ... the remaining setup is unchanged
+
+    # get the containing directory of this file
+    # use $BATS_TEST_FILENAME instead of ${BASH_SOURCE[0]} or $0,
+    # as those will point to the bats executable's location or the preprocessed file respectively
+    CURRDIR="$( cd "$( dirname "$BATS_TEST_FILENAME" )" >/dev/null 2>&1 && pwd )"
+
+    # make executables visible to PATH
+    PATH="$CURRDIR/../bin:$PATH"
```

Comment:
hub extensions dir should also be added to the PATH:
PATH="$CURRDIR/../bin:$CURRDIR/../:$PATH" with assumption that user runs the tests from within $CURRDIR/tests

Reply:
CURRDIR falls into the `BATS_TEST_FILENAME` (current filename) `files` `dotenv` are in `../bin`